### PR TITLE
handle kids in different forms; add local ion resolver

### PR DESCRIPTION
**Pull request by decentralgabe**

### CODE REVIEW

File: did/util_test.go [hunk 1]
```diff
@@ -199,6 +199,32 @@ func TestGetKeyFromVerificationInformation(t *testing.T) {
...
+
+		key, err := GetKeyFromVerificationMethod(docWithHash, "test-did#test-kid")
+		assert.NoError(t, err)
+		assert.Equal(t, pubKey, key)
+
+		docWithHash.VerificationMethod[0].ID = "test-kid"
+		docWithoutHash := docWithHash
+		key, err = GetKeyFromVerificationMethod(docWithoutHash, "test-did#test-kid")
+		assert.NoError(t, err)
+		assert.Equal(t, pubKey, key)
```
Comment (andresuribe87) [inline did/util_test.go:225]: Separate this to a different sub-test for consistency.

Comment (andresuribe87) [inline did/util_test.go:225]: Separate this to a different sub-test for consistency.

File: did/util.go [hunk 2]
```diff
@@ -62,12 +62,22 @@ func GetKeyFromVerificationMethod(did Document, kid string) (gocrypto.PublicKey,
 
 // matchesKIDConstruction checks if the targetID matches possible combinations of the did and kid
 func matchesKIDConstruction(did, kid, targetID string) bool {
-	maybeKID1 := kid                            // the kid == the kid
-	maybeKID2 := fmt.Sprintf("#%s", kid)        // the kid == the fragment with a #
-	maybeKID3 := fmt.Sprintf("%s#%s", did, kid) // the kid == the DID ID + the fragment with a #
-	maybeKID4 := fmt.Sprintf("%s%s", did, kid)  // the kid == the DID ID + the fragment without a #
-	maybeKID5, found := strings.CutPrefix(kid, did)
-	return targetID == maybeKID1 || targetID == maybeKID2 || targetID == maybeKID3 || targetID == maybeKID4 || (found && targetID == maybeKID5)
+	maybeKID1 := kid                                // the kid == the kid
```
Comment (andresuribe87) [inline did/util.go:65]: Cascading conditions make more sense now that there are 7 guesses. You can do that with a switch

**Review by andresuribe87 — APPROVED** [review]
Guessing what the intentions of the JWT signer are instead of making them explicit makes me think that this can be exploited in some way. I can't think of exactly how, but there's something off about this.

File: did/util_test.go (hunk 1 above)
Comment (decentralgabe) [reply did/util_test.go:225]: done

File: did/util.go (hunk 2 above)
Comment (decentralgabe) [reply did/util.go:65]: I tried that but found it much less readable...

**Pull request merged**